pkg/ingestor/parser/cdx_vex/parser_cdx_vex.go

```diff
@@ -0,0 +1,217 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdx_vex
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	cdx "github.com/CycloneDX/cyclonedx-go"
+
+	"github.com/guacsec/guac/pkg/assembler"
+	"github.com/guacsec/guac/pkg/assembler/clients/generated"
+	"github.com/guacsec/guac/pkg/assembler/helpers"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/ingestor/parser/common"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+var vexStatusMap = map[cdx.ImpactAnalysisState]generated.VexStatus{
+	"resolved":     generated.VexStatusFixed,
+	"exploitable":  generated.VexStatusAffected,
+	"in_triage":    generated.VexStatusUnderInvestigation,
+	"not_affected": generated.VexStatusNotAffected,
+}
+
+var justificationsMap = map[cdx.ImpactAnalysisJustification]generated.VexJustification{
+	"code_not_present":   generated.VexJustificationVulnerableCodeNotPresent,
+	"code_not_reachable": generated.VexJustificationVulnerableCodeNotInExecutePath,
+}
+
+type cdxVexParser struct {
+	doc               *processor.Document
+	identifierStrings *common.IdentifierStrings
+	cdxBom            *cdx.BOM
+}
+
+func NewCdxVexParser() common.DocumentParser {
+	return &cdxVexParser{
+		identifierStrings: &common.IdentifierStrings{},
+	}
+}
+
+// Parse breaks out the document into the graph components
+func (c *cdxVexParser) Parse(ctx context.Context, doc *processor.Document) error {
+	c.doc = doc
+	err := json.Unmarshal(doc.Blob, &c.cdxBom)
+	if err != nil {
+		return fmt.Errorf("unable to unmarshal CDX VEX document: %w", err)
+	}
+
+	return nil
+}
+
+// GetIdentities gets the identity node from the document if they exist
+func (c *cdxVexParser) GetIdentities(ctx context.Context) []common.TrustInformation {
+	return nil
+}
+
+func (c *cdxVexParser) GetIdentifiers(ctx context.Context) (*common.IdentifierStrings, error) {
+	return nil, fmt.Errorf("not yet implemented")
```

Comment:
can add `Identifiers` similar to [cdx parser](https://github.com/guacsec/guac/blob/main/pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go#L41C2-L41C19)